### ISSUE TITLE
[Synthese] SQL - non mise à jours des croisements spatiaux

### ISSUE THREAD

**Issue by jbrieuclp** [PnX-SI/GeoNature#739]
Lorsqu'un relevé est modifié, le trigger `pr_occtax.fct_tri_synthese_update_releve` doit mettre à jour la table de synthese en conséquence, ça c'est ok ça fonctionne. 
Ensuite un autre trigger  lance la fonction `gn_synthese.fct_trig_insert_in_cor_area_synthese()` pour choper les géométries intersectées par le relevé (ST_Intersects). Dans cette fonction il y a une vérification, si update, de la modification de la geométrie (`the_geom_local`) pour refaire ou pas le croisement. Impec ! Seulement the_geom_local n'est jamais mis à jour entre le relevé et la synthese dans le premier trigger, seuls `the_geom_4326` et `the_geom_point` sont mis à jour, du coup la mise à jours des intersections ne s’exécute pas au modification de relevés.

Faudrait ajouter la ligne `the_geom_local = NEW.geom_local,` pour que ça roule.

C'est là que ça se passe :
https://github.com/PnX-SI/GeoNature/blob/cc83d88e403c1cfee9d860cc9034804c08d63e6c/data/migrations/2.0.0rc4.2to2.0.0.sql#L710-L746

**Comment by camillemonchicourt** [comment]
Exact, problème aussi noté ici : https://github.com/PnX-SI/GeoNature/issues/727

Et une description des triggers dans la doc : http://docs.geonature.fr/admin-manual.html#triggers-vers-la-synthese

**Comment by jbrieuclp** [comment]
Du coup faudrait intégrer la modif de la fonction `pr_occtax.fct_tri_synthese_update_releve()` avec la ligne `the_geom_local = NEW.geom_local,` dans l'UPDATE dans le prochain migration.sql.

**Comment by jbrieuclp** [comment]
En lien, pour la fonction `gn_synthese.fct_trig_insert_in_cor_area_synthese()` au niveau de la vérification du changement de géométrie ou non [L321](https://github.com/PnX-SI/GeoNature/blob/cc83d88e403c1cfee9d860cc9034804c08d63e6c/data/migrations/2.1.0to2.1.1.sql#L321). Si un des deux champs `OLD.the_geom_local, NEW.the_geom_local` est NULL le retour est NULL et non boolean. Un COALESCE va carriger ça : `COALESCE(public.ST_EQUALS(OLD.the_geom_local, NEW.the_geom_local), false)`
Même si les 2 geoms sont null, au pire le st_intersects qui suit sera exécuté mais ne sera pas trop coûteux du fait que la geométrie est vide.
